File: 71. Simplify Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        temp = [word for word in path.strip().split('/') if word and word != '.']
        ls = []
        for word in temp:
            if word == '..':
                if ls: ls.pop(-1)
            else:
                ls.append(word)
        return '/'+'/'.join(ls)

File: 71. Simplify Path.java
class Solution {
    public String simplifyPath(String path) {
        String[] part = path.split("/");
        Deque<String> stack = new ArrayDeque<>();
        for (String s : part) {
            if (s.equals("") || s.equals(".")) {
                continue;
            } else if (s.equals("..")) {
                if (stack.isEmpty()) {
                    continue;
                } else {
                    stack.pop();
                }
            } else {
                stack.push(s);
            }
        }
        StringBuilder sb = new StringBuilder();
        sb.append("/");
        while (!stack.isEmpty()) {
            sb.append(stack.pollLast());
            if (!stack.isEmpty()) {
                sb.append("/");
            }
        }
        return sb.toString();
    }
}